### ISSUE TITLE
lower mAP results with ADAM optimization - training custom dataset

### ISSUE THREAD

**Issue by buyukkanber** [ultralytics/yolov5#8234]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello,

I have been training my custom dataset with YOLOv5x model. When I train the dataset with selecting **SGD** optimizer the mAP 0.5, 0.95 score is **_0.769_**, but if I train the same dataset with only changing optimizer to **ADAM** the result is _**0.664**_ mAP value of mAP 0.5, 0.95 score.

What makes ADAM optimizer lower the mAP score?

Thanks

**Comment by AyushExel** [comment]
@buyukkanber there are a lot of factors that can cause performance differences between the optimizers. Adam generally converges faster when all of it parameters are adequately set but these are hyper params and the optimal value can change due to any number of reasons(batch size, dataset, epochs etc.). Which is why optimizers themselves can be considered as hyper-parameters. Did you customize the optimizer params manually?

**Comment by buyukkanber** [comment]
No, I did not. I only added `--optimizer Adam` to the training command

**Comment by glenn-jocher** [comment]
@buyukkanber Understood. The Adam optimizer is known for its adaptiveness to learning rates, which can lead to faster convergence on some datasets. You might have to fine-tune the learning rate and other hyperparameters specific to Adam to achieve better performance. 🚀 You can also experiment with different learning rates and epsilon values for Adam to see if it improves your results. Let me know if you need further assistance!
